bn_mp_radix_size_overestimate.c

```diff
@@ -0,0 +1,258 @@
+       183550ul, 1343271782ul, 2148848508ul, 1429396043ul, 3785381171ul,
+    638097011ul, 1073048670ul, 1651143338ul, 3473298766ul,          0ul
+};
+
+/* TODO: This assumes the type "long long" exists. If it doesn't: use the tables for MP_16BIT */
+#elif ( (defined MP_28BIT) || (defined MP_31BIT) || (defined MP_32BIT)  || (defined MP_64BIT) )
+#define LTM_RADIX_SIZE_SCALE 61
+#ifdef MP_64BIT
+static const mp_digit logbases[65] = {
+#else
+static const mp_word logbases[65] = {
+#endif
+                      0ull,                    0ull,  2305843009213693952ull,
+    3654674702153732339ull,  4611686018427387904ull,  5354001665492895103ull,
+    5960517711367426291ull,  6473319721408181094ull,  6917529027641081856ull,
+    7309349404307464679ull,  7659844674706589055ull,  7976906213687625599ull,
+    8266360720581120243ull,  8532633055092329421ull,  8779162730621875046ull,
+    9008676367646627443ull,  9223372036854775808ull,  9425047617917838031ull,
+    9615192413521158631ull,  9795053960520401544ull,  9965687683920283007ull,
+   10127994423561913434ull, 10282749222901319551ull, 10430623713119086168ull,
+   10572203729794814195ull, 10708003330985790206ull, 10838476064306023373ull,
+   10964024106461197019ull, 11085005739835568998ull, 11201741516507896395ull,
+   11314519376860321395ull, 11423598928577852074ull, 11529215046068469760ull,
+   11631580915841357939ull, 11730890627131531983ull, 11827321386901076197ull,
+   11921035422734852583ull, 12012181624960263419ull, 12100896969734095496ull,
+   12187307757246061761ull, 12271530693133976959ull, 12353673836347420459ull,
+   12433837432775607386ull, 12512114650772492370ull, 12588592232115013503ull,
+   12663351069800359783ull, 12736466722332780120ull, 12808009872697200099ull,
+   12878046739008508147ull, 12946639442816362188ull, 13013846340199484158ull,
+   13079722320071570371ull, 13144319073519717325ull, 13207685337486676934ull,
+   13269867115674890971ull, 13330907879180520702ull, 13390848749049262950ull,
+   13449728662674133884ull, 13507584525721590347ull, 13564451351070528819ull,
+   13620362386074015347ull, 13675349229302017118ull, 13729441937791546026ull,
+   13782669125715645774ull, 13835058055282163712ull
+};
+#else
+#  ifdef _MSC_VER
+#    pragma message("mp_radix_size_overestimate: unknown or undefined MP_xBIT type")
+#  else
+#    warning "mp_radix_size_overestimate: unknown or undefined MP_xBIT type"
+#  endif
+#endif
+/* *INDENT-ON* */
+
+mp_err mp_radix_size_overestimate(const mp_int *a, const int base, int *size)
+{
+   mp_int bi_bit_count, bi_k;
+#if ( (defined MP_8BIT) || (defined MP_16BIT) )
+   mp_int bi_k_bis;
+#endif
+   int rem, bit_count;
+   mp_err e = MP_OKAY;
+
+   if ((base < 2) || (base > 64)) {
+      *size = 0;
+      return MP_VAL;
+   }
+   /* floor( log_2(a) ) + 1 */
+   bit_count = mp_count_bits(a);
+
+   if (bit_count == 0) {
+      *size = 2;
+      return MP_OKAY;
+   }
+   if (base == 2) {
```

Comment:
Here you always add one for the sign. Why special case for 2? Should be handled by power of two code.

Reply:
> Here you always add one for the sign

No, it is the plus one in `floor( log_2(a) ) + 1 ` and plus one for `\0`

> Why special case for 2? Should be handled by power of two code.

Yes, that's right, needs to be merged.

Reply:
Then you are missing the sign?

Reply:
> Here you always add one for the sign

No I didn't add the one for the sign at all for base 2, that was a bug you found!
Thanks!

Reply:
Why am I missing push-notes (at least with Firefox)? This website does not update regularily, The "Then your are missing the sign" came by email first, the "same here" came as a a push-note immediately!
Do I have to change to "Edge"? ;-)